text/0022-response-context.md

```diff
@@ -0,0 +1,98 @@
+
+# Background
+
+Since the `Response` metadata is PII sensitive, it should be properly mapped, documented and scrubbed.
+
+# Supporting Data
+
+See [Supporting data](https://www.notion.so/sentry/Failed-HTTP-Client-requests-automatically-result-in-Events-f6c21d2a58ce4f2c889a823fd1da0044#0ca951d5216742dbaab02f5fd33b8fb5) section in the [DACI](https://www.notion.so/sentry/Failed-HTTP-Client-requests-automatically-result-in-Events-f6c21d2a58ce4f2c889a823fd1da0044).
+
+# Options Considered
+
+## Option 1 (Preferred)
+
+Adding a `Response` interface directly in the Event payload.
+
+```json
+{
+  "response": {
+    "method": "POST",
+    "url": "http://absolute.uri/foo",
+    "query_string": "query=foobar&page=2",
+    "data": { // arbitrary fields
+      "foo": "bar"
+    },
+    "cookies": "PHPSESSID=298zf09hf012fh2; csrftoken=u32t4o3tb3gg43; _gat=1;",
+    "headers": {
+      "content-type": "text/html"
+    },
+    "env": {
+      "REMOTE_ADDR": "192.168.0.1"
+    },
+    "status_code": 500,
+    "status_message": "foo bar", // is that useful?
+    "is_redirect": false,
+    "response_body_size": 1000, // bytes
+  }
+}
+```
+
+The `data` field is semantically different than the `Request#data`. The `Request#data` is the data that was sent to the server, while the `Response#data` is just arbitrary data that is attached to the response. The `Response#data` is not used for anything in the SDKs, but it is useful for the user to attach arbitrary data to the response.
+We could rename to `other` or something else to avoid confusion.
+
+## Option 2
+
+Adding a `Response` interface in the [Contexts interface](https://develop.sentry.dev/sdk/event-payloads/contexts/).
+
+```json
+{
+    "contexts": {
+        "response": {
+            "type": "response"
+            // ...
+  }
+}
+```
+
+The content is the same as in Option 1.
+
+## Option 3
+
+Expand the `Request` interface adding the missing fields.
+Data scrubbing should consider response headers when scrubbing.
+If we do that, the `Request` docs should be ammended that it contains the `Response` data as well otherwise it's semantically wrong.
+
+## Must have for all the options
+
+A tag should be created for `url` and `status_code` fields, people should be able to search for events with a specific `url` or `status_code`, also to alert on specific `status_code`.
+
+# Drawbacks
+
+The `Response` interface is PII sensitive, so we need to be careful about how we scrub it.
+
+The [Dart Dio HTTP Client](https://docs.sentry.io/platforms/dart/configuration/integrations/dio/) integration already adds the `Response` field in the `Contexts` interface, so in case we go with Option 1, we'd need to migrate the data to the new `Response` interface.
+
+# Unresolved questions
+
+* Is `status_message` useful?
+* Should we rename `data` to `other` or something else to avoid confusion?
+* Are there any other fields that should be added to the `Response` interface?
```

Comment:
After addressing https://github.com/getsentry/rfcs/pull/22/files#r986505011 this becomes less relevant.